api/app/controllers/review.controller.ts

```diff
@@ -441,3 +443,71 @@ export const getAllReviewsForAdmin = async (req: Request, res: Response) => {
     }
   }
 };
+
+export const resetReviews = async (req: Request, res: Response) => {
+  const { id } = req.params; // Get the review ID from the request parameters
+
+  try {
+    // Find the review
+    const review = await Review.findOne({ where: { id } });
+
+    if (!review) {
+      return res.status(404).send("Review not found");
+    }
+
+    // Reset the transcript
+    await Transcript.update(
+      { status: "queued", claimedBy: null },
+      { where: { id: review.transcriptId } }
+    );
+
+    // Get the updated transcript
+    const updatedTranscript = await Transcript.findOne({
+      where: { id: review.transcriptId },
+    });
+
+    if (!updatedTranscript) {
+      throw new Error("Transcript not found");
+    }
+    // Now you can use updatedTranscript.id
+
+    // Delete the review
+    await Review.destroy({ where: { id } });
+
+    // Clear the Redis cache for the review
+    redis.del(`review:${id}`, (err, succeeded) => {
+      if (err) {
+        throw err;
+      }
+      Logger.info(`Redis cache cleared for review ${id}: ${succeeded}`);
+    });
+
+    // Clear the Redis cache for the transcript
+    redis.del(`transcript:${updatedTranscript.id}`, (err, succeeded) => {
+      if (err) {
+        throw err;
+      }
+      Logger.info(
+        `Redis cache cleared for transcript ${updatedTranscript.id}: ${succeeded}`
+      );
+    });
+
+    // Clear the Redis cache for the pages of transcripts
+    const totalItems = await Transcript.count();
+    const totalPages = Math.ceil(totalItems / DB_QUERY_LIMIT);
+    for (let page = 1; page <= totalPages; page++) {
+      redis.del(`transcripts:page:${page}`, (err, succeeded) => {
+        if (err) {
+          throw err;
+        }
+        Logger.info(
+          `Redis cache cleared for transcripts page ${page}: ${succeeded}`
+        );
+      });
+    }
```

Comment:
This logic seems to already exist here:
https://github.com/bitcointranscripts/transcription-review-backend/blob/9207455055b2111a46dc359a36f430702fa99a05/api/app/controllers/transcript.controller.ts#L325-L331

and here:
https://github.com/bitcointranscripts/transcription-review-backend/blob/9207455055b2111a46dc359a36f430702fa99a05/api/app/controllers/transcript.controller.ts#L396-L402

Maybe we could have this logic in a re-usable function?
Also based on the code that I linked to, it seems that on other occasions we are also deleting another extra cache (`await redis.srem("cachedTranscripts", id);`)

Reply:
Yeah, but we might still need to tweak the Redis flow later for each controller, I will advise we still leave it as is.